lib/useOnyx.ts

```diff
@@ -75,23 +75,35 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(
     const connectionRef = useRef<Connection | null>(null);
     const previousKey = usePrevious(key);
 
+    const currentDependenciesRef = useRef(dependencies);
+    currentDependenciesRef.current = dependencies;
+
+    const currentSelectorRef = useRef(options?.selector);
+    currentSelectorRef.current = options?.selector;
+
     // Create memoized version of selector for performance
     const memoizedSelector = useMemo(() => {
         if (!options?.selector) return null;
 
         let lastInput: OnyxValue<TKey> | undefined;
         let lastOutput: TReturnValue;
+        let lastDependencies: DependencyList = [];
         let hasComputed = false;
 
         return (input: OnyxValue<TKey> | undefined): TReturnValue => {
-            // Always recompute when input changes
-            if (!hasComputed || lastInput !== input) {
-                const newOutput = options.selector!(input);
+            const currentDependencies = currentDependenciesRef.current;
+            const currentSelector = currentSelectorRef.current;
+
+            // Recompute if input changed, dependencies changed, or first time
+            const dependenciesChanged = !deepEqual(lastDependencies, currentDependencies);
```

Comment:
Quick question, if I use dynamic onyx key, should I add the dynamic part to the dependencies list?

ie:
```javascript
    const [transaction] = useOnyx(`${ONYXKEYS.COLLECTION.TRANSACTION}${getNonEmptyStringOnyxID(transactionID)}`, {
        canBeMissing: true,
    });
```

Reply:
Do you think that's expensive to do a deepEqual here every time this memo is being calculated?

I saw there's some `useOnyx` usages that have the entire collection as dependencies.

Reply:
> Do you think that's expensive to do a deepEqual here every time this memo is being calculated?

Depending on what the dependencies are - for primitive values it should be fine, but things might be definitely expensive for large objects. Hopefully there is no need of passing entire collections as dependencies.

> From here, I think I have another question. What's the best practice for selectors?
>
> Should it be pure function? If a selector has a large object as a dependency, should we avoid using inline function?

Yes, they definitely should be pure functions. We should avoid inline selectors to avoid re-renders coming from new function references on each render.